### PR TITLE
Refactor mesh shading code

### PR DESCRIPTION
**Pull request by panxinmiao**
To enhance the reusability of the code, facilitate the expansion of the current Mesh shading properties, and allow for the addition of more shading models in the future, it is necessary to refactor the current Mesh shading code. 

The main objectives of this refactor are:

1. Extract the Punctual light-related lighting logic for reuse across different shading models.
2. Use pointers for passing lights to avoid extensive parameter copying.
3. Temporarily move other Material-related shading properties and logic to the main file of the Mesh fragment shader. In the future, consider extracting more granular reusable code segments and using these segments to construct shading logic for each Mesh material (utilizing separate WGSL files as the framework for the material shading logic).

### CODE REVIEW

**Review by almarklein — APPROVED** [review]
IIUC none of the logic has changed; just moving code around, right?

**Comment by panxinmiao** [comment]
> IIUC none of the logic has changed; just moving code around, right?

Yes, there are no any changes to the shading logic.

**Comment by almarklein** [comment]
That's the validation examples doing what they're for 😉

File: pygfx/renderers/wgpu/wgsl/light_pbr.wgsl [hunk 1]
```diff
@@ -100,83 +100,42 @@ fn computeMultiscattering(normal: vec3<f32>, view_dir: vec3<f32>, specular_color
 }
 
 fn RE_IndirectSpecular_Physical(radiance: vec3<f32>, irradiance: vec3<f32>,
-        geometry: GeometricContext, material: PhysicalMaterial, reflected_light: ReflectedLight) -> ReflectedLight{
+        geometry: GeometricContext, material: PhysicalMaterial, reflected_light: ptr<function, ReflectedLight>){
```
Comment (Korijn) [inline pygfx/renderers/wgpu/wgsl/light_pbr.wgsl:103]: I had to look up the pointer type definition syntax to understand this, I thought it was some kind of to-from mapping (function that returns a type). But instead, `function` means the data is stored in the function's address space. Is that correct?

**Review by Korijn — APPROVED** [review]

**Comment by Korijn** [comment]
@almarklein please hit the merge button if you are also OK with these changes!

File: pygfx/renderers/wgpu/wgsl/light_pbr.wgsl (hunk 1 above)
Comment (panxinmiao) [reply pygfx/renderers/wgpu/wgsl/light_pbr.wgsl:103]: Yes, it is a local variable, using memory locations in the "function" address space.

See:  https://www.w3.org/TR/WGSL/#ref-ptr-types

**Comment by almarklein** [comment]
Thanks @panxinmiao !

**Pull request merged**